sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/physical/partitioning.scala

```diff
@@ -262,6 +260,15 @@ case class RangePartitioning(ordering: Seq[SortOrder], numPartitions: Int)
     super.satisfies0(required) || {
       required match {
         case OrderedDistribution(requiredOrdering) =>
+          // If `ordering` is a prefix of `requiredOrdering`:
+          //   - Let's say `ordering` is [a, b] and `requiredOrdering` is [a, b, c]. If a row is
+          //     larger than another row w.r.t. [a, b], it's also larger w.r.t. [a, b, c]. So
+          //     `RangePartitioning(a, b)` satisfies `OrderedDistribution(a, b, c)`.
+          //
+          // If `requiredOrdering` is a prefix of `ordering`:
+          //   - Let's say `ordering` is [a, b, c] and `requiredOrdering` is [a, b]. If a row is
```

Comment:
"If a row is ... satisfies ..." => According to the RangePartitioning definition, any [a1, b1, c1] in a previous partition must be smaller than any [a2, b2, c2] in the following partition, which means any row in the previous partition must have either 1) [a1, b1] smaller than [a2, b2]; or 2) [a1, b1] equal to [a2, b2] and c1 smaller than c2. So `RangePartitioning(a, b, c)` satisfies `OrderedDistribution(a, b)` which requires any [a1, b1] from a previous partition smaller than any [a2, b2] from a following partition."